### PR TITLE
Update Allocation from Fleet Documentation

### LINKED ISSUE #3759: Update Allocation from a Fleet Documentation
The Allocation from a Fleet documentation uses the deprecated required. https://github.com/googleforgames/agones/blob/65a997e5185afb1592797957a7bbcdf29226ec41/site/content/en/docs/Integration%20Patterns/allocation-from-fleet.md?plain=1#L19-L26 We should update the example to use selectors similar to the Game Server Allocation reference https://github.com/googleforgames/agones/blob/65a997e5185afb1592797957a7bbcdf29226ec41/site/content/en/docs/Reference/gameserverallocation.md?plain=1#L28-L40

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3759 

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Integration Patterns/allocation-from-fleet.md [hunk 1]
```diff
@@ -20,9 +20,12 @@ label selector to target a specific `Fleet` by name. In this instance, we are ta
...
 spec:
-  required:
-    matchLabels:
-      agones.dev/fleet: xonotic
+  selectors:
+    - matchLabels:
+        agones.dev/fleet: xonotic
+      players:
+        minAvailable: 0
+        maxAvailable: 99
```
Comment (igooch) [inline site/content/en/docs/Integration Patterns/allocation-from-fleet.md:28]:
```suggestion
```

**Review by igooch — COMMENTED** [review]
One comment, otherwise LGTM.

File: site/content/en/docs/Integration Patterns/allocation-from-fleet.md (hunk 1 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Integration Patterns/allocation-from-fleet.md]: I have applied your suggestion.

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**